jetstream/analysis.py

```diff
@@ -670,17 +889,44 @@ def validate(self, use_glean_ids: bool = False) -> None:
             use_glean_ids=use_glean_ids,
         )
 
-        self._write_sql_output(
-            f"enrollments_{bq_normalize_name(self.config.experiment.normandy_slug)}",
-            enrollments_sql,
-        )
+        self._write_sql_output(f"enrollments_{bq_normalize_name(experiment_slug)}", enrollments_sql)
 
         dry_run_query(enrollments_sql)
-        print(f"Dry running enrollments query for {self.config.experiment.normandy_slug}:")
-        print(enrollments_sql)
+        logger.info(f"Dry running enrollments query for {experiment_slug}:")
+        logger.info(enrollments_sql)
+
+        if not metric_slugs:
+            output_loc = f"metrics_{bq_normalize_name(experiment_slug)}"
+            logger.debug(f"Dry running metrics query for {experiment_slug}")
+            self.validate_metric_query(exp, metrics, limits, output_loc, use_glean_ids)
+        else:
+            selected_metrics = [m for m in metrics if m.name in metric_slugs]
+            for i, metric in enumerate(selected_metrics):
+                output_loc = f"metrics_{bq_normalize_name(experiment_slug)}_{metric.name}"
+                logger.debug(
+                    f"Dry running metric [{metric.name}] query for {experiment_slug}"
+                    f"({i + 1} of {len(metric_slugs)})"
+                )
+                self.validate_metric_query(exp, [metric], limits, output_loc, use_glean_ids)
```

Comment:
Just a thought: if this step ends up taking long, we could also run the dry runs in parallel

Reply:
Good idea, I think I can just decorate `validate_metric_query` with `@dask.delayed` and get this automatically. Could be helpful if you're running multiple metrics with `--metric-slug`.